handler.h

```diff
@@ -112,6 +112,19 @@ class NotFoundHandler : public RequestHandler {
 
 REGISTER_REQUEST_HANDLER(NotFoundHandler);
 
+class BlockingHandler : public RequestHandler {
+public:
```

Comment:
I might not be understanding something, but why do we need a BlockingHandler? What is the purpose of this?